src/java.base/share/classes/jdk/internal/module/SystemModuleFinders.java

```diff
@@ -442,14 +450,25 @@ private InputStream toInputStream(ByteBuffer bb) { // ## -> ByteBuffer?
             }
         }
 
+        /**
+         * Returns the node for the given resource if found. If the name references
+         * a non-resource node, then an {@link Optional#empty() empty optional} is
+         * returned even if a non-resource node exists with the given name.
+         */
+        private Optional<ImageReader.Node> findResourceNode(ImageReader reader, String name) throws IOException {
```

Comment:
If you change this method use SystemImage.reader (like containsResource does) then it would be a bit nicer. It's okay to change this method to return null when not found too, the read method can box.